js/ui/components/clubhouse.js

```diff
@@ -209,23 +209,17 @@ var ClubhouseComponent = new Lang.Class({
     _overrideAddNotification: function() {
         this._oldAddNotificationFuncPrototype =
             NotificationDaemon.GtkNotificationDaemon.prototype.AddNotificationAsync;
-        NotificationDaemon.GtkNotificationDaemon.prototype.ClubhouseData = [CLUBHOUSE_ID,
-                                                                            this.source,
-                                                                            this._oldAddNotificationFuncPrototype];
-        NotificationDaemon.GtkNotificationDaemon.prototype.AddNotificationAsync = function(params, invocation) {
+        NotificationDaemon.GtkNotificationDaemon.prototype.AddNotificationAsync = (params, invocation) => {
```

Comment:
Hmm, I had that whole hacky way of passing the data because I noticed that the `this` inside the prototype's body referred to the `GtkNotificationDaemon`. Maybe the using the arrow func changes that?

Reply:
Yes, arrow functions are a better way of doing the Lang.bind() dance. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#No_separate_this